File: PrelimLabWork3/Java/PrelimLabWork3.java
import java.util.Scanner;

public class PrelimLabWork3 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.println("==== Required Prelim Exam Score Calculator =====");

        // Input
        System.out.print("Enter Attendance score: ");
        double attendance = sc.nextDouble();

        System.out.print("Enter Lab Work 1 score: ");
        double lab1 = sc.nextDouble();

        System.out.print("Enter Lab Work 2 score: ");
        double lab2 = sc.nextDouble();

        System.out.print("Enter Lab Work 3 score: ");
        double lab3 = sc.nextDouble();

        // Compute Lab Work Average
        double labAverage = (lab1 + lab2 + lab3) / 3;

        // Compute Class Standing (30% of Prelim Grade)
        double classStanding = (attendance * 0.40) + (labAverage * 0.60);

        // Compute required Prelim Exam scores for target grades
        // Prelim Grade = (Prelim Exam * 0.70) + (Class Standing * 0.30)
        double requiredPassing = (75 - (classStanding * 0.30)) / 0.70;
        double requiredExcellent = (100 - (classStanding * 0.30)) / 0.70;

        // Output
        System.out.println("\n===== RESULTS =====");
        System.out.printf("Attendance: %.2f%n", attendance);
        System.out.printf("Lab Work 1: %.2f%n", lab1);
        System.out.printf("Lab Work 2: %.2f%n", lab2);
        System.out.printf("Lab Work 3: %.2f%n", lab3);
        System.out.printf("Lab Work Average: %.2f%n", labAverage);
        System.out.printf("Class Standing: %.2f%n", classStanding);

        System.out.println("\nRequired Prelim Exam Scores:");
        if (requiredPassing > 100) {
            System.out.println("To PASS (75): Impossible based on current Class Standing.");
        } else if (requiredPassing <= 0) {
            System.out.println("To PASS (75): You are already passing even with 0 exam score!");
        } else {
            System.out.printf("To PASS (75): %.2f%n", requiredPassing);
        }

        if (requiredExcellent > 100) {
            System.out.println("To get EXCELLENT (100): Impossible.");
        } else {
            System.out.printf("To get EXCELLENT (100): %.2f%n", requiredExcellent);
        }

        sc.close();
    }
}

File: PrelimLabWork3/JavaScript/PrelimLabWork3.js
function compute() {
    let attendance = parseFloat(document.getElementById("attendance").value);
    let lab1 = parseFloat(document.getElementById("lab1").value);
    let lab2 = parseFloat(document.getElementById("lab2").value);
    let lab3 = parseFloat(document.getElementById("lab3").value);

    if (isNaN(attendance) || isNaN(lab1) || isNaN(lab2) || isNaN(lab3)) {
        alert("Please enter all values correctly.");
        return;
    }

    let labAvg = (lab1 + lab2 + lab3) / 3;
    let classStanding = (attendance * 0.40) + (labAvg * 0.60);

    let requiredPassing = (75 - (classStanding * 0.30)) / 0.70;
    let requiredExcellent = (100 - (classStanding * 0.30)) / 0.70;

    let out = `<div class="section-title">Results</div>`;
    out += `<p>Attendance: ${attendance.toFixed(2)}</p>`;
    out += `<p>Lab Work 1: ${lab1.toFixed(2)}</p>`;
    out += `<p>Lab Work 2: ${lab2.toFixed(2)}</p>`;
    out += `<p>Lab Work 3: ${lab3.toFixed(2)}</p>`;
    out += `<p>Lab Work Average: ${labAvg.toFixed(2)}</p>`;
    out += `<p>Class Standing: ${classStanding.toFixed(2)}</p>`;

    out += `<div class="section-title">Required Prelim Exam Scores</div>`;

    if (requiredPassing > 100) {
        out += `<p>To PASS (75): Impossible based on current Class Standing.</p>`;
    } else if (requiredPassing <= 0) {
        out += `<p>To PASS (75): Already guaranteed even with 0 exam score.</p>`;
    } else {
        out += `<p>To PASS (75): ${requiredPassing.toFixed(2)}</p>`;
    }

    if (requiredExcellent > 100) {
        out += `<p>To get EXCELLENT (100): Impossible based on current Class Standing.</p>`;
    } else if (requiredExcellent <= 0) {
        out += `<p>To get EXCELLENT (100): Already guaranteed even with 0 exam score.</p>`;
    } else {
        out += `<p>To get EXCELLENT (100): ${requiredExcellent.toFixed(2)}</p>`;
    }

    document.getElementById("output").innerHTML = out;
}

function clearFields() {
    document.getElementById("attendance").value = "";
    document.getElementById("lab1").value = "";
    document.getElementById("lab2").value = "";
    document.getElementById("lab3").value = "";
    document.getElementById("output").innerHTML = "";
}
